TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -79,341 +119,484 @@ namespace TEN::Entities::Creatures::TR3
+				creature.Enemy = nullptr;
+
+			GetCreatureMood(&item, &AI, true);
 
-			if (creature->Enemy == LaraItem &&
+			if (creature.Enemy == LaraItem &&
 				AI.distance > CIVVY_ESCAPE_RANGE &&
 				AI.enemyFacing < ANGLE(67.0f) &&
 				AI.enemyFacing > -ANGLE(67.0f))
 			{
-				creature->Mood = MoodType::Escape;
+				creature.Mood = MoodType::Escape;
 			}
 
-			CreatureMood(item, &AI, true);
+			CreatureMood(&item, &AI, true);
 
-			angle = CreatureTurn(item, creature->MaxTurn);
+			angle = CreatureTurn(&item, creature.MaxTurn);
 
-			auto* realEnemy = creature->Enemy;
-			creature->Enemy = LaraItem;
 
-			if ((laraAI.distance < CIVVY_AWARE_RANGE || item->HitStatus || TargetVisible(item, &laraAI)) &&
-				!(item->AIBits & FOLLOW))
+			//TODO: Make a way to detect entities if it has an AI Object (A current issue for other entities too).
+			if ( item.HitStatus ||
+				 (targetDistance <= CIVVY_AWARE_RANGE && 
+				  creature.Enemy != nullptr && TargetVisible(&item, &AI) &&
+				  !(item.AIBits & FOLLOW))
+				)
 			{
-				if (!creature->Alerted)
-					SoundEffect(SFX_TR3_AMERCAN_HOY, &item->Pose);
+				if (!creature.Alerted)
+					SoundEffect(SFX_TR3_AMERCAN_HOY, &item.Pose);
 
 				AlertAllGuards(itemNumber);
 			}
-			creature->Enemy = realEnemy;
 
-			switch (item->Animation.ActiveState)
+			switch (item.Animation.ActiveState)
 			{
-			case CIVVY_WAIT:
-				if (creature->Alerted || item->Animation.TargetState == CIVVY_STATE_RUN_FORWARD)
-				{
-					item->Animation.TargetState = CIVVY_STATE_IDLE;
-					break;
-				}
+				case CIVVY_STATE_IDLE:
 
-			case CIVVY_STATE_IDLE:
-				creature->MaxTurn = 0;
-				creature->Flags = 0;
-				extraHeadRot.y = laraAI.angle;
+					creature.MaxTurn = 0;
+					creature.Flags = 0;
 
-				if (item->AIBits & GUARD)
-				{
-					extraHeadRot.y = AIGuard(creature);
-					if (!(GetRandomControl() & 0xFF))
+					if (AI.ahead)
+						jointHeadRot.y = targetAngle;
+
+					if (item.AIBits & GUARD)
+					{
+						jointHeadRot.y = AIGuard(&creature);
+
+						if (!(GetRandomControl() & 0xFF))
+						{
+							if (item.Animation.ActiveState == CIVVY_STATE_IDLE)
+								item.Animation.TargetState = CIVVY_STATE_WAIT;
+							else
+								item.Animation.TargetState = CIVVY_STATE_IDLE;
+						}
+
+						break;
+					}
+					else if (item.AIBits & PATROL1)
+						item.Animation.TargetState = CIVVY_STATE_WALK;
+					else if (creature.Mood == MoodType::Escape)
+					{
+						if (Lara.TargetEntity != &item && AI.ahead)
+							item.Animation.TargetState = CIVVY_STATE_IDLE;
+						else
+							item.Animation.TargetState = CIVVY_STATE_RUN;
+					}
+					else if (creature.Mood == MoodType::Bored ||
+						(item.AIBits & FOLLOW && (creature.ReachedGoal || targetDistance > SQUARE(BLOCK(2)))))
 					{
-						if (item->Animation.ActiveState == CIVVY_STATE_IDLE)
-							item->Animation.TargetState = CIVVY_WAIT;
+						if (!creature.Alerted && Random::TestProbability(CIVVY_WAIT_CHANCE))
```

Comment:
There should have `item.RequiredState` based on the official source:
```
if (item.Animation.RequiredState)
item.Animation.TargetState = item.Animation.RequiredState;
```

Reply:
Yes, I removed it because this entity is not calling RequiredStates anymore. 
I could add that check to keep similarities with other entities, but is unused here now.

Formerly, it was using the RequireState to activate the Wait animation (the one where he shakes his punches), but I found the logic some unusual so I reworked it a bit. now Walk sends to IDLE and while is in IDLE, if he's bored then he randomly  will do its wait animation.

Reply:
okay i approve it then :)